### ISSUE TITLE
Bug report: an error occurred while list files using the object storage interface

### ISSUE THREAD

**Issue by mervinkid** [chubaofs/chubaofs#421]
**Describe the bug**
Using ChubaoFS as storage backend on Apache Flink environment, an error occurred while Flink was performing the List Files operation.

Exception respond by Apache Flink:
```
Caused by: org.apache.flink.fs.s3base.shaded.com.amazonaws.services.s3.model.AmazonS3Exception: Invalid Argument (Service: Amazon S3; Status Code: 400; Error Code: InvalidArgument; Request ID: bbfe064947f5422e8ddbf57ef76a63d8; S3 Extended Request ID: null), S3 Extended Request ID: null
 at org.apache.flink.fs.s3base.shaded.com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1639)
 at org.apache.flink.fs.s3base.shaded.com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1304)
 at org.apache.flink.fs.s3base.shaded.com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1056)
```

**Environment**
SDK version in http header traced by **ObjectNode**:
```
aws-sdk-java/1.11.271 Linux/3.10.0-693.5.2.el7.jd_3313.x86_64 Java_HotSpot(TM)_64-Bit_Server_VM/25.131-b11 java/1.8.0_131 scala/2.11.12, presto
```

**To Reproduce**
List files using object storage interface with multi-level directory as prefix.

Example:
The following file is stored on the ChubaoFS volume.
```
dir1 / dir1_1 / file_1_1_1.txt
```
Then invoke ListObjects interface with `dir1/dir1` as prefix.

**Expected behavior**
None.

**Additional context**
None.

**Comment by mervinkid** [comment]
Issue resolved by PR #422.

**Comment by mervinkid** [comment]
A new problem was found when there was data that exactly matched the prefix, then the results were not consistent with the expected results.
This issue is not completely resolved, reopen this issue.
